addons/godot-next/references/array_2d.gd

```diff
@@ -180,6 +190,13 @@ func fill_col(p_idx: int, p_value):
 		arr.push_back(p_value)
 	set_col(p_idx, arr)
 
+func flatten() -> Array:
+	var out = []
+	for a_row in range(len(data)):
+		for a_col in range(len(data[a_row])):
+			out.append({"location" : Vector2(a_row, a_col), "content" : data[a_row][a_col]})
+	return out
+
```

Comment:
I feel that the term "flatten" for the name is misleading given that it not only converts the data from a 2D array to a 1D array, but it *also* transforms the structure of the elements, namely, converting them from whatever-data-type-they-are to a Dictionary with key-value pairs that users are expected to know.

Rather than displaying the location with every element, I think you should add another method, say, `flattened_offset(Vector2) -> int` that calculates what the 1D offset would be given a Vector2 and leave the elements as they are in the flattened array. That way, the structure of the data doesn't change and people still have an established API for getting the information they need.

May also consider having a separate class for Array-specific methods that could have the flattened offset thing. Cause, at that point, the calculation is more relevant to 1D arrays than the Array2D class (it would just need the Array, the presumed height, and the intended Vector2). You also don't want someone to call `flatten()`, make changes to the new array, and then call `flattened_offset` in the Array2D while expecting to get an accurate result (different data, so it will give a different result now).

Reply:
Yes, I'm using these function in my [Roguelike Starter Kit](https://github.com/verbalshadow/rlsk) in the Dungeon Generation Algorithms.

I can rewrite flatten() to be a true flatten like you said. As written it allows me easy access to the content and still know where that value was in the array2d. Which I'm not sure how you would have the true flat array. If you have a better name for this version let me know.

 flattened_offset() as/in a separate class doesn't seem like it would work cause you need access to the array2D size.  

```
func flattened_offset(pos : Vector2) -> int:
    return pos.x + pos.y * len(data[0])
```

Which I guess you could add as an argument. To bad there isn't a way to attach the function to Array itself.

Reply:
Ping. Maybe have `get_flattened()` for just returning the data in a 1D array and a `get_flat_map()` for returning a Dictionary that contains Vector2 keys and Variant values. I also think converting that/those names from verbs to getters is better since they aren't actually mutating the data in any way.

Reply:
Another ping @verbalshadow.

Reply:
@verbalshadow Are you still planning on fixing up this pull request?

Reply:
@willnationsdev I think that the purpose of this class got simply misinterpreted (`Array2D` to contain `Vector2`-only values, which is not the case, it would be the case if the class was named `Vector2Array2D`), I'd suggest the author to create a dedicated class for that, if still interested.